client/messaging/message-input.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useRef, useState } from 'react'
+import styled from 'styled-components'
+import KeyListener from '../keyboard/key-listener'
+import TextField from '../material/text-field'
+import { colorDividers } from '../styles/colors'
+
+const StyledTextField = styled(TextField)<{ showDivider?: boolean }>`
+  flex-shrink: 0;
+  position: relative;
+  padding: 8px 16px;
+  contain: content;
+
+  &::after {
+    position: absolute;
+    height: 1px;
+    left: 0px;
+    right: 0px;
+    top: 0;
+
+    content: '';
+    border-top: 1px solid ${props => (props.showDivider ? colorDividers : 'transparent')};
+    transition: border 250ms linear;
+  }
+`
+
+export interface MessageInputProps {
+  className?: string
+  showDivider?: boolean
+  onSendChatMessage: (msg: string) => void
+}
+
+export default function MessageInput(props: MessageInputProps) {
+  const [message, setMessage] = useState<string>('')
+  const inputRef = useRef<TextField | null>(null)
+
+  const onChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target
+      if (value !== message) {
```

Comment:
useState hooks already do this check, no need to do it here (and it will make it so this callback doesn't get recreated on nearly every render anyway, which is also nice :) )